### PR TITLE
adding s3 policy for chef artifacts; adding global iam policies

### PR DESCRIPTION
**Pull request by malkolm**

### CODE REVIEW

File: tyr/servers/server.py [hunk 1]
```diff
@@ -545,6 +550,8 @@ def resolve_iam_role(self):
         self.log.info('Existing policies: '
                       '{policies}'.format(policies=existing_policies))
 
+        self.IAM_ROLE_POLICIES.extend(self.GLOBAL_IAM_ROLE_POLICIES)
+        self.IAM_ROLE_POLICIES = set(self.IAM_ROLE_POLICIES)
```
Comment (Z3r0Sum) [inline tyr/servers/server.py]: @citruspi @malkolm - What's the advantage of using a set here over a list?

**Comment by Z3r0Sum** [comment]
@malkolm Looks good to me.

File: tyr/servers/server.py (hunk 1 above)
Comment (citruspi) [inline tyr/servers/server.py]:
```
>>> my_list=list([5, 5, 6, 6, 7])
>>> my_list
[5, 5, 6, 6, 7]
>>> my_set=set([5, 5, 6, 6, 7])
>>> my_set
set([5, 6, 7])
>>> 
```

It basically removes any duplicates so we don't re-process policies again.

File: tyr/servers/server.py [hunk 2]
```diff
@@ -21,6 +21,10 @@ class Server(object):
     NAME_SEARCH_PREFIX = '{envcl}-{location}-'
     NAME_AUTO_INDEX = True
 
+    GLOBAL_IAM_ROLE_POLICIES = ['allow-get-chef-artifacts-chef-client',
```
Comment (citruspi) [inline tyr/servers/server.py]: Could you format the list with entries and brackets on separate lines? [Like this][0].

[0]: https://github.com/hudl/Tyr/blob/master/tyr/servers/solr/slave.py#L9

**Comment by citruspi** [comment]
@malkolm Aside from the formatting of the list, it looks good.

File: tyr/servers/server.py (hunk 2 above)
Comment (malkolm) [inline tyr/servers/server.py]: done

File: tyr/servers/server.py (hunk 1 above)
Comment (Z3r0Sum) [inline tyr/servers/server.py]: ah, cool

**Pull request merged**